geoana/gravity.py

```diff
@@ -0,0 +1,410 @@
+        >>> import matplotlib.pyplot as plt
+        >>> from geoana.gravity import PointMass
+
+        Define the point mass.
+
+        >>> location = np.r_[0., 0., 0.]
+        >>> mass = 1.0
+        >>> simulation = PointMass(
+        >>>     mass=mass, location=location
+        >>> )
+
+        Now we create a set of gridded locations and compute the gravitational gradient.
+
+        >>> X, Y = np.meshgrid(np.linspace(-1, 1, 20), np.linspace(-1, 1, 20))
+        >>> Z = np.zeros_like(X) + 0.25
+        >>> xyz = np.stack((X, Y, Z), axis=-1)
+        >>> g_tens = simulation.gravitational_gradient(xyz)
+
+        Finally, we plot the gravitational gradient for each element of the 3 x 3 matrix.
+
+        >>> fig = plt.figure(figsize=(10, 10))
+        >>> gs = fig.add_gridspec(3, 3, hspace=0, wspace=0)
+        >>> (ax1, ax2, ax3), (ax4, ax5, ax6), (ax7, ax8, ax9) = gs.subplots(sharex='col', sharey='row')
+        >>> fig.suptitle('Gravitational Gradients')
+        >>> ax1.contourf(X, Y, g_tens[:,:,0,0])
+        >>> ax2.contourf(X, Y, g_tens[:,:,0,1])
+        >>> ax3.contourf(X, Y, g_tens[:,:,0,2])
+        >>> ax4.contourf(X, Y, g_tens[:,:,1,0])
+        >>> ax5.contourf(X, Y, g_tens[:,:,1,1])
+        >>> ax6.contourf(X, Y, g_tens[:,:,1,2])
+        >>> ax7.contourf(X, Y, g_tens[:,:,2,0])
+        >>> ax8.contourf(X, Y, g_tens[:,:,2,1])
+        >>> ax9.contourf(X, Y, g_tens[:,:,2,2])
+        >>> plt.show()
+        """
+
+        r_vec = xyz - self.location
+        r = np.linalg.norm(r_vec, axis=-1)
+        g_tens = -G * self.mass * (np.eye(3) / r[..., None, None] ** 3 -
+                                   3 * r_vec[..., None] * r_vec[..., None, :] / r[..., None, None] ** 5)
+        return g_tens
+
+
+class Sphere(PointMass):
+    """Class for gravitational solutions for a sphere.
+
+    The ``Sphere`` class is used to analytically compute the gravitational
+    potentials, fields, and gradients for a sphere.
+
+    Parameters
+    ----------
+    density : float
+        Density of sphere (m).  Default is r = 1 kg/m^3.
+    radius : float
+        Radius of sphere (m).  Default is r = 1 m.
+    mass : float
+        Mass of the sphere (kg). Default is m = 4/3*pi*R^3*rho kg.
+    location : array_like, optional
+        Center of the sphere (m). Default is (0, 0, 0).
+    """
+
+    def __init__(self, density=1.0, radius=1.0, mass=1.0, location=None, **kwargs):
+        super().__init__(mass, location, **kwargs)
+        self.radius = radius
+        self.density = density
+
+    @property
+    def radius(self):
+        """Radius of the sphere in meters
+
+        Returns
+        -------
+        float
+            Radius of the sphere in meters
+        """
+        return self._radius
+
+    @radius.setter
+    def radius(self, item):
+        item = float(item)
+        if item < 0.0:
+            raise ValueError('radius must be non-negative')
+        self._radius = item
+
+    @property
+    def density(self):
```

Comment:
In most areas of this package we have named things by there common mathematical symbol, so here it would be, `rho` and then describe it as density in the docstrings as you have done so nicely here.